commerce/domain/orchestrators/classes/CartCalculateExecutorForUnitTest.cls

```diff
@@ -10,31 +10,54 @@ global class CartCalculateExecutorForUnitTest extends CartExtension.CartCalculat
 
     global override void prices(CartExtension.CartCalculateCalculatorRequest request) {
         CartExtension.Cart cart = request.getCart();
-        cart.setName('CartRepriced');
+
+        String nameToAppend = shouldCallExternalPricingService(cart)
```

Comment:
I'm not a fan of such use-case as an public example. Recommended approach would be to apply such logic in the calculator pseudocode:
```
class MyCalculator extends Calculator {
   void calculate(CartCalculationRequest request) {
      CartExtension.Cart cart = request.getCart();
      if (cart.getStatus() == 'Active') {
         super.calculate(request);//default internal pricing
      } else {
         //custom implementation using BuyerActionDetails and external callout
      }
   }
```
}